dataikuapi/dss/ml.py

```diff
@@ -1046,88 +1084,130 @@ def get_modality_data(self, definition):
 
 
-class DSSSubpopulationAnalyses(DSSInternalDict):
+class DSSSubpopulationAnalyses(object):
     """
     Object to read details of subpopulation analyses of a trained model
 
     Do not create this object directly, use :meth:`DSSTrainedPredictionModelDetails.get_subpopulation_analyses()` instead
     """
 
     def __init__(self, data, prediction_type):
-        super(DSSSubpopulationAnalyses, self).__init__(data)
+        if data is None:
+            self._internal_dict = dict()
+        else:
+            self._internal_dict = data
         self.prediction_type = prediction_type
         self.analyses = []
         for analysis in data.get("subpopulationAnalyses", []):
             self.analyses.append(DSSSubpopulationAnalysis(analysis, prediction_type))
-    
+
+    def get_raw(self):
+        """
+        Gets the raw dictionary of the actual data
+
+        :rtype: dict
+        """
+        return self._internal_dict
+
+    def __repr__(self):
+        return self.__class__.__name__ + "(" + self._internal_dict.__repr__() + ")"
+
     def get_global(self):
         """
         Retrieves information and performance on the full dataset used to compute the subpopulation analyses
         """
-        return DSSSubpopulationGlobal(self.get("global"), self.prediction_type)
+        return DSSSubpopulationGlobal(self._internal_dict["global"], self.prediction_type)
 
     def list_analyses(self):
         """
         Lists all features on which subpopulation analyses have been computed
         """
-        return [analysis.get("feature") for analysis in self.analyses]
+        return [analysis.get_raw()["feature"] for analysis in self.analyses]
     
     def get_analysis(self, feature):
         """
         Retrieves the subpopulation analysis for a particular feature
         """
         try:
-            return next(analysis for analysis in self.analyses if analysis.get("feature") == feature)
+            return next(analysis for analysis in self.analyses if analysis.get_raw()["feature"] == feature)
         except StopIteration:
             raise ValueError("Subpopulation analysis for feature '%s' cannot be found" % feature)
 
 
-class DSSPartialDependence(DSSInternalDict):
+class DSSPartialDependence(object):
     """
     Object to read details of partial dependence of a trained model
 
     Do not create this object directly, use :meth:`DSSPartialDependencies.get_partial_dependence(feature)` instead
     """
 
     def __init__(self, data):
-        super(DSSPartialDependence, self).__init__(data)
+        if data is None:
+            self._internal_dict = dict()
+        else:
+            self._internal_dict = data
+
+    def get_raw(self):
+        """
+        Gets the raw dictionary of the actual data
+
+        :rtype: dict
+        """
+        return self._internal_dict
+
+    def __repr__(self):
+        return self.__class__.__name__ + "(" + self._internal_dict.__repr__() + ")"
```

Comment:
Ditto → column name + type?

Reply:
I couldn't find the type for this class, only the feature.